pkg/logs/config/messages.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://wwm.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package config
+
+import (
+	"strings"
+	"sync"
+)
+
+// Messages holds a message message
+type Messages struct {
+	messages map[string]string
+	lock     *sync.Mutex
+}
+
+// NewMessages initialize Messages with the default values
+func NewMessages() *Messages {
+	return &Messages{
+		messages: make(map[string]string),
+		lock:     &sync.Mutex{},
+	}
+}
+
+// AddMessage create a message
+func (m *Messages) AddMessage(key string, message string) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	m.messages[key] = message
+}
+
+// AddWarning create a warning
+func (m *Messages) AddWarning(key string, warning string) {
+	m.AddMessage("warning_"+key, warning)
```

Comment:
Why are you adding a prefix here ? If it's just for filtering out purpose, I'd rather avoid doing that because a string compare is not efficient.
You could create a structure:
```
type Message struct {
   Value String
   IsWarning bool
}
```
And use IsWarning for filtering out

Reply:
It's to avoid collision between keys and to reuse the same methods so most warning methods are just calling the message one with the prefix `warning_`